docs/src/modules/java-protobuf/pages/workflows.adoc

```diff
@@ -0,0 +1,200 @@
+
+[source,java,indent=0]
+.src/main/java/com/example/transfer/api/TransferWorkflow.java
+----
+include::example$java-protobuf-transfer-workflow/src/main/java/com/example/transfer/api/TransferWorkflow.java[tag=definition]
+----
+<1> Each step should have a unique name.
+<2> We instruct Kalix to run a given deferred call, to withdraw funds from a wallet.
+<3> After successful withdrawal we return an `Effect` that will update the workflow state and move to the next step called `"deposit"`. An input parameter for this step is a `Deposit` message.
+<4> Another workflow step action to deposit funds to a given wallet.
+<5> This time we return an effect that will stop workflow processing, by using special `end` method.
+<6> We collect all steps to form a workflow definition.
+
+== Retrieving state
+
+To have access to the current state of the workflow we can use `currentState()` or the command handler parameter. However, if this is the first command we are receiving for this workflow, the state will be `null`. We can change it by overriding the `emptyState` method. The following example shows the implementation of the read-only command handler:
+
+[source,java,indent=0]
+.src/main/java/com/example/transfer/api/TransferWorkflow.java
+----
+include::example$java-protobuf-transfer-workflow/src/main/java/com/example/transfer/api/TransferWorkflow.java[tag=get-transfer]
+----
+<1> Returns the current state as reply for the request.
+
+
+IMPORTANT: For simplicity purposes, we are returning the internal state directly back to the requester. In a real-world scenario, it's usually better to instead convert this internal domain model into a public model so the internal representation is free to evolve without breaking clients code.
+
+// TODO uncomment in next PR A full transfer workflow source code is available https://github.com/lightbend/kalix-jvm-sdk/tree/main/samples/java-protobuf-transfer-workflow[here, {tab-icon}, window="new"]. Follow the `README` file to run and test it.
+
+== Pausing workflow
+
+A long-running workflow can be paused while waiting for some additional information to continue processing. A special `pause` transition can be used to inform Kalix that the execution of the Workflow should be postponed. By launching a request to a Workflow endpoint, the user can then resume the processing. Additionally, a Kalix Timer can be scheduled to automatically inform the Workflow that the expected time for the additional data has passed.
+
+
+[source,java,indent=0]
+.src/main/java/com/example/transfer/api/TransferWorkflow.java
+----
+include::example$java-protobuf-transfer-workflow-compensation/src/main/java/com/example/transfer/api/TransferWorkflow.java[tag=pausing]
+----
+<1> Schedules a timer as a Workflow step action. Make sure that the timer name is unique for every Workflow instance.
+<2> Pauses the Workflow execution.
+
+NOTE: Remember to cancel the timer once the Workflow is resumed. Also, adjust the Workflow timeout to match the timer schedule.
```

Comment:
`Also, adjust the Workflow timeout to match the timer schedule. ` This is mentioned before showing Workflow's timeouts. Maybe add a link?

Reply:
added link